src/Adapters/Storage/Mongo/MongoTransform.js

```diff
@@ -487,10 +487,14 @@ function transformTopLevelAtom(atom) {
       return BytesCoder.JSONToDatabase(atom);
     }
     if (GeoPointCoder.isValidJSON(atom)) {
-      return GeoPointCoder.JSONToDatabase(atom);
+      // normally we transform a 'GeoPoint' to just the name when storing to the db, but
+      // _GlobalConfig has no schema so we must preserve type information and other fields
+      return className === "_GlobalConfig" ? atom : GeoPointCoder.JSONToDatabase(atom);
```

Comment:
sure, at the very least I can move this logic outside of this function but I get its still not ideal to have this component care about the className